erdpy/config.py

```diff
@@ -162,10 +165,56 @@ def get_defaults() -> Dict[str, Any]:
 
 
 def read_file() -> Dict[str, Any]:
-    if not os.path.isfile(CONFIG_PATH):
-        return dict()
-    return utils.read_json_file(CONFIG_PATH)
+    if os.path.isfile(LOCAL_CONFIG_PATH):
+        return utils.read_json_file(LOCAL_CONFIG_PATH)
+    if os.path.isfile(CONFIG_PATH):
+        return utils.read_json_file(CONFIG_PATH)
+    return dict()
 
 
 def write_file(data: Dict[str, Any]):
-    utils.write_json_file(CONFIG_PATH, data)
+    if os.path.isfile(LOCAL_CONFIG_PATH):
+        utils.write_json_file(LOCAL_CONFIG_PATH, data)
+    else:
+        utils.write_json_file(CONFIG_PATH, data)
+
+
+def add_config_args(argv):
+    try:
+        func, subcommand, *_ = argv
+    except ValueError:
+        return argv
+
+    config = read_file()
+
+    try:
+        config_args = config[func][subcommand]
+    except KeyError:
+        return argv
+
+    extra_args = []
+    for key, value in config_args.items():
+        key_arg = f'--{key}'
+        # arguments from the command line override the config
+        if key_arg in argv:
+            continue
+        extra_args.append(key_arg)
+        if value is True:
+            continue
+        if isinstance(value, List):
+            for item in value:
+                extra_args.append(str(item))
+        else:
+            extra_args.append(str(value))
+
+    # the verbose flag is an exception since it has to go before the command and subcommand
+    # eg. erdpy --verbose contract deploy
+    verbose_flag = '--verbose'
+    pre_args = []
+    if verbose_flag in extra_args:
+        extra_args.remove(verbose_flag)
+        pre_args = [verbose_flag]
+
+    final_args = pre_args + argv + extra_args
```

Comment:
Would it be possible to operate directly on the `args` object instead? It might be easier to simply inject the local config values into the `args` object instead of modifying `argv` before parsing.

Or better yet, if possible, `parser.set_defaults(...)` might read the values from the local config file?

Reply:
I tried using `set_defaults`, but this fails with something along the lines:
`erdpy contract COMMAND [-h] ... deploy: error: the following arguments are required: --pem, --keyfile, --ledger, --nonce, --gas-limit`
At the moment the creation of the parser is in a circular dependency with the arguments that are given from the command line. This is because we want some arguments to be required, but we may provide one of several (eg. we need a wallet which we can specify either with `--pem` or `--keyfile` and at least one of them is required). Because of this circular dependency the `set_defaults` doesn't work, as it adds the default arguments, but requires a parser to have been built in the first place.

This could (probably?) be rewritten by using [`ArgumentParse.add_mutually_exclusive_group`](https://docs.python.org/3/library/argparse.html#mutual-exclusion) which should eliminate the circular dependency.
Here, my first thought was to add the extra args handling but keep the rest of the code unchanged, in order to not break anything. The refactoring should be done in another PR probably.